src/fastpath/fastpath.c

```diff
@@ -472,6 +811,71 @@ void NORETURN fastpath_reply_recv(word_t cptr, word_t msgInfo)
     callerSlot->cteMDBNode = nullMDBNode;
 #endif
 
+#ifdef CONFIG_EXCEPTION_FASTPATH
+    if (unlikely(fault_type != seL4_Fault_NullFault)) {
```

Comment:
Why are these special handling for these fault types needed? The default is restart=1 for all other fault types?

Reply:
They have some extra logic as the other types of exceptions may replace parts of the register set of the faulting thread. Right now, these are not necessary as we are only handling vm faults but they're more like stubs to expand upon later down the line.